sql/src/main/java/io/crate/executor/transport/RerouteActions.java

```diff
@@ -73,6 +77,39 @@ private RerouteActions() {
         return listener;
     }
 
+    public static CompletableFuture<Long> executeRetryFailed(
+        BiConsumer<ClusterRerouteRequest, ActionListener<ClusterRerouteResponse>> rerouteAction) {
+
+        ClusterRerouteRequest request = new ClusterRerouteRequest();
+        request.setRetryFailed(true);
+        FutureActionListener<ClusterRerouteResponse, Long> listener =
+            new FutureActionListener<>(resp -> {
```

Comment:
Might make sense to extract this logic into a static function and reference it using a method ref. For readability - but would also make it easier to test it.